lib/spack/spack/build_environment.py

```diff
@@ -812,11 +812,6 @@ def child_process(child_pipe, input_stream):
                 setup_package(pkg, dirty=dirty)
             return_value = function()
             child_pipe.send(return_value)
-        except StopIteration as e:
-            # StopIteration is used to stop installations
-            # before the final stage, mainly for debug purposes
-            tty.msg(e)
-            child_pipe.send(None)
```

Comment:
Instead of getting rid of this, keep it and propagate the new exception type (see below).